hudi-common/src/main/java/org/apache/hudi/avro/MercifulJsonConverter.java

```diff
@@ -187,196 +178,774 @@ private static Object convertJsonToAvroField(Object value, String name, Schema s
+
+    private static JsonToAvroFieldProcessor generateDoubleTypeHandler() {
+      return new JsonToAvroFieldProcessor() {
+        @Override
+        public Pair<Boolean, Object> convert(Object value, String name, Schema schema, boolean shouldSanitize, String invalidCharMask) {
+          if (value instanceof Number) {
+            return Pair.of(true, ((Number) value).doubleValue());
+          } else if (value instanceof String) {
+            return Pair.of(true, Double.valueOf((String) value));
+          }
+          return Pair.of(false, null);
+        }
+      };
+    }
+
+    private static JsonToAvroFieldProcessor generateFloatTypeHandler() {
+      return new JsonToAvroFieldProcessor() {
+        @Override
+        public Pair<Boolean, Object> convert(Object value, String name, Schema schema, boolean shouldSanitize, String invalidCharMask) {
+          if (value instanceof Number) {
+            return Pair.of(true, ((Number) value).floatValue());
+          } else if (value instanceof String) {
+            return Pair.of(true, Float.valueOf((String) value));
+          }
+          return Pair.of(false, null);
+        }
+      };
+    }
+
+    private static JsonToAvroFieldProcessor generateLongTypeHandler() {
+      return new JsonToAvroFieldProcessor() {
+        @Override
+        public Pair<Boolean, Object> convert(Object value, String name, Schema schema, boolean shouldSanitize, String invalidCharMask) {
+          if (value instanceof Number) {
+            return Pair.of(true, ((Number) value).longValue());
+          } else if (value instanceof String) {
+            return Pair.of(true, Long.valueOf((String) value));
+          }
+          return Pair.of(false, null);
+        }
+      };
+    }
+
+    private static JsonToAvroFieldProcessor generateStringTypeHandler() {
+      return new JsonToAvroFieldProcessor() {
+        @Override
+        public Pair<Boolean, Object> convert(Object value, String name, Schema schema, boolean shouldSanitize, String invalidCharMask) {
+          return Pair.of(true, value.toString());
+        }
+      };
+    }
+
+    private static JsonToAvroFieldProcessor generateBytesTypeHandler() {
+      return new JsonToAvroFieldProcessor() {
+        @Override
+        public Pair<Boolean, Object> convert(Object value, String name, Schema schema, boolean shouldSanitize, String invalidCharMask) {
+          // Should return ByteBuffer (see GenericData.isBytes())
+          return Pair.of(true, ByteBuffer.wrap(value.toString().getBytes()));
+        }
+      };
+    }
+
+    private static JsonToAvroFieldProcessor generateFixedTypeHandler() {
+      return new JsonToAvroFieldProcessor() {
+        @Override
+        public Pair<Boolean, Object> convert(Object value, String name, Schema schema, boolean shouldSanitize, String invalidCharMask) {
+          // The ObjectMapper use List to represent FixedType
+          // eg: "decimal_val": [0, 0, 14, -63, -52] will convert to ArrayList<Integer>
```

Comment:
Interesting, is this as per specification? I am curious why not List<Long>?

Reply:
this is day 1 implementation, it is marked as new code change due to indentation is changed. We should check with the original owner of the code.

Reply:
ok if it's not changing behavior